Src/java/engine/src/main/java/org/opencds/cqf/cql/engine/execution/CqlEngine.java

```diff
@@ -315,4 +372,1416 @@ private Set<String> getExpressionSet(Library library) {
+
+        return AnyInValueSetEvaluator.internalEvaluate(codes, anyInValueSet.getValueset(), valueset, state);
+    }
+
+    @Override
+    public Object visitInValueSet(InValueSet inValueSet, State state) {
+
+        Object code = visitExpression(inValueSet.getCode(), state);
+        Object vs = null;
+        if (inValueSet.getValueset() != null) {
+            vs = ValueSetRefEvaluator.toValueSet(state, inValueSet.getValueset());
+        } else if (inValueSet.getValuesetExpression() != null) {
+            vs = visitExpression(inValueSet.getValuesetExpression(), state);
+        }
+        return InValueSetEvaluator.inValueSet(code, vs, state);
+    }
+
+    @Override
+    public Object visitValueSetRef(ValueSetRef elm, State state) {
+        return ValueSetRefEvaluator.internalEvaluate(state, elm);
+    }
+
+    @Override
+    public Object visitXor(Xor elm, State state) {
+        Object left = validateOperand(visitExpression(elm.getOperand().get(0), state));
+        Object right = validateOperand(visitExpression(elm.getOperand().get(1), state));
+        return XorEvaluator.xor(left, right);
+    }
+
+
+    @Override
+    public Object visitWidth(Width elm, State state) {
+        Object operand = validateOperand(visitExpression(elm.getOperand(), state));
+        return WidthEvaluator.width(operand);
+    }
+
+    @Override
+    public Object visitVariance(Variance variance, State state) {
+        Object source = visitExpression(variance.getSource(), state);
+        return VarianceEvaluator.variance(source, state);
+    }
+
+    @Override
+    public Object visitAvg(Avg avg, State state) {
+        Object src = visitExpression(avg.getSource(), state);
+        return AvgEvaluator.avg(src, state);
+    }
+
+    @Override
+    public Object visitDivide(Divide elm, State state) {
+        Object left = validateOperand(visitExpression(elm.getOperand().get(0), state));
+        Object right = validateOperand(visitExpression(elm.getOperand().get(1), state));
+        return DivideEvaluator.divide(left, right, state);
+    }
+
+
+    @Override
+    public Object visitUpper(Upper elm, State state) {
+        Object operand = validateOperand(visitExpression(elm.getOperand(), state));
+        return UpperEvaluator.upper(operand);
+    }
+
+    @Override
+    public Object visitUnion(Union elm, State state) {
+        Object left = validateOperand(visitExpression(elm.getOperand().get(0), state));
+        Object right = validateOperand(visitExpression(elm.getOperand().get(1), state));
+        return UnionEvaluator.union(left, right, state);
+    }
+
+    @Override
+    public Object visitGreater(Greater elm, State state) {
+        try {
+            Object left = validateOperand(visitExpression(elm.getOperand().get(0), state));
+            Object right = validateOperand(visitExpression(elm.getOperand().get(1), state));
+
+            return GreaterEvaluator.greater(left, right, state);
+        } catch (Exception e) {
```

Comment:
we should move exception handling up to a central location. All the visit methods should be trivial.